solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -1679,19 +1698,21 @@ public static String getCollectionPath(String coll) {
    * @see ZkStateReader#unregisterCore(String)
    */
   public void registerCore(String collection) {
-    AtomicBoolean reconstructState = new AtomicBoolean(false);
+    AtomicReference<StateWatcher> newWatcherRef = new AtomicReference<>();
     collectionWatches.compute(
         collection,
         (k, v) -> {
           if (v == null) {
-            reconstructState.set(true);
-            v = new StatefulCollectionWatch();
+            StateWatcher stateWatcher = new StateWatcher(collection);
```

Comment:
Can we replace this with
can we replace this with
```
 v = new StatefulCollectionWatch(new StateWatcher(collection));
 newWatcherRef.set(v.associatedWatcher);
```

Reply:
Curious, is there a reason to prefer this beyond style? Style-wise I somewhat prefer _not_ writing to a volatile variable and then reading back from it ... but because the `StatefulCollectionWatch` instance is not accessible from other threads at this point there's no functional issue with it.